spec/lib/granite/action_spec.rb

```diff
@@ -7,6 +7,55 @@
     end
   end
 
+  describe 'Dependency Injection' do
```

Comment:
I added a couple comments to individual tests, but tbh I really don't see value in any of these tests. You're testing basic ruby method overloading in subclass, nothing here seems granite specific.

Reply:
@andriusch This is to prevent regressions.

I want to ensure that the documentation is aligned with the code.
I want to ensure that if we change attributes internals, the documented code is still working and we don't accidentally override this behavior.

Reply:
But we can't override it outside of very aggressive shenanigans. Any subclass in Ruby can override `initialize` method and redefine what it does and define `attr_reader`s. Any method in Ruby can set instance variables.

The only 2 ways these tests could break due to changes in granite is:
- if we redefine what `attr_reader` does in Granite::Action.
- if we refefine what `private` does in Granite::Action.

And none of those sounds like something that would be allowed to merge.

Reply:
@andriusch I'm more interested to prevent `active_data` regressions than `granite` ones.

That means, the tests are to protect us `active_data` changing the way attributes are handled. This isn't in `granite` as you're pointing. 🙂 

I agree with you that we won't do any of those changes in `granite`.

Given, I already removed [some controversial specs](https://github.com/toptal/granite/pull/64/commits/014e1e20cdf896f9b7d5c2eeb5e8f9cdb3d661b3), what's your take on this? To remove this `describe` block?

Reply:
Yeah I'd suggest removing the whole describe. If your concern is `active_data`, I'm pretty sure we already have specs covering this, e.g. https://github.com/toptal/granite/blob/master/spec/lib/granite/action/subject_spec.rb#L32. `active_data` itself seems to be under maintenance mode so it's unlikely that they'll have any breaking API changes. TBH I think something needs to be done regarding `active_data`, but that's another story.

Reply:
Removed https://github.com/toptal/granite/pull/64/commits/281a985a8cd2ccb32d9e4ed5305e3f97baf4aa9e